readme.md

```diff
@@ -0,0 +1,176 @@
+# Tech Spec - Teleport Challenge
+
+## Overview
+To be implemented:
+
+1. `Job` library for starting, stopping, and getting status and streamed logs of jobs.
+2. API that implements the `Job` library, permitting clients to call the 4 public methods.
+3. CLI Client that calls API, using command line args to specify API methods.
+
+## Job Library
+`Job` type will include public fields:
+
+- Command
+- Args
+- ProcessID
+
+##### Implement methods for:
+
+- Start
+	- Assure job command exists. Executables can be located in Go using `exec.LookPath()`.
+	- Assure that a duplicate command is not currently running by checking "DB".
```

Comment:
If duplicate detection is to be implemented, how is this check not racy? The only way to do this would be to use some atomic lock/acquire check at this stage to prevent two racing job start requests.